clang/test/Driver/sycl-post-link-options.cpp

```diff
@@ -29,3 +29,52 @@
 // RUN:   --sycl-post-link-options="-O2 -device-globals -O0" \
 // RUN:   --linker-path=/usr/bin/ld %t.o -o a.out 2>&1 | FileCheck --check-prefix OPTIONS_POSTLINK_JIT_NEW_SYCLBIN %s
 // OPTIONS_POSTLINK_JIT_NEW_SYCLBIN: sycl-post-link{{.*}} -spec-const=native -properties -split=auto -emit-only-kernels-as-entry-points -emit-param-info -symbols -emit-kernel-names -emit-exported-symbols -emit-imported-symbols -split-esimd -lower-esimd -O2 -device-globals -O0
+//
+// Ensure driver forwards these triple based options to clang-linker-wrapper.
+// 
+// RUN: %clangxx %s -### -fsycl --offload-new-driver \
+// RUN:   -fsycl-remove-unused-external-funcs \
+// RUN:   -fsycl-device-code-split-esimd \
+// RUN:   -fsycl-add-default-spec-consts-image \
+// RUN:   2>&1 | FileCheck --check-prefix=OPTIONS_FORWARD %s
+// OPTIONS_FORWARD: clang-linker-wrapper{{.*}} "-sycl-remove-unused-external-funcs" "-sycl-device-code-split-esimd" "-sycl-add-default-spec-consts-image"
+//
+// RUN: %clangxx %s -### -fsycl --offload-new-driver \
```

Comment:
What is the behavior if `--offload-new-driver` is not passed?

Reply:
These have been tested in other tests:

-fsycl-device-code-split-esimd: https://github.com/intel/llvm/blob/fcd35c06ce1e421f01d3d6b096fd57ca7ff57d6b/clang/test/Driver/sycl-offload-with-split-old-model.c#L307-L319

-fsycl-remove-unused-external-funcs: https://github.com/intel/llvm/blob/fcd35c06ce1e421f01d3d6b096fd57ca7ff57d6b/clang/test/Driver/sycl-offload-old-model.cpp#L149-L154

-fsycl-add-default-sepc-consts-image: https://github.com/intel/llvm/blob/fcd35c06ce1e421f01d3d6b096fd57ca7ff57d6b/clang/test/Driver/sycl-add-default-spec-consts-image-old-model.cpp#L1-L23